app/src/main/java/com/mandysoftware/livrenouveau/SearchActivity.kt

```diff
@@ -0,0 +1,274 @@
+                        )
+                    )
+                }
+                return true
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+
+                if (di != null) {
+                    di.postCommand(
+                        CompositeDictionary.Command(
+                            CompositeDictionary.CommandName.SUGGEST, newText
+                        )
+                    )
+                }
+                return true
+            }
+        })
+
+        //searchView.setFocusable(true)
+        searchView.setIconified(false)
+        //searchView.requestFocusFromTouch()
+        //searchView.requestFocus()
+    }
+
+    override fun onDictionaryError(command: CompositeDictionary.Command?, err: Exception) {
+        println("ERROR: " + command.toString() + " Message: " + err.message.toString())
+    }
+
+    override fun onDictionaryDownloadProgress(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        bytesDownloaded: Int,
+        bytesTotal: Int
+    ) {
+
+    }
+
+    override fun onDictionaryDownloadComplete(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+
+    }
+
+    override fun onDictionaryLoad(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+
+    }
+
+    override fun onDictionaryDelete(
+        dictionaryInfo: CompositeDictionary.DictionaryInfo,
+        success: Boolean,
+        err: Exception?
+    ) {
+
+    }
+
+    override fun onAllDictionariesReady(success: Boolean, err: Exception?) {
+
+    }
+
+    override fun onIndexDownloadProgress(bytesDownloaded: Int, bytesTotal: Int) {
+
+    }
+
+    override fun onIndexDownloadComplete(success: Boolean, err: Exception?) {
+
+    }
+
+    override fun onHeadFilesComplete(
+        successCount: Int,
+        totalCount: Int,
+        success: Boolean,
+        err: Exception?
+    ) {
+
+    }
+
+    override fun onLocalFilesChecked() {
+
+
+    }
+
+    override fun onDictionaryLookup(
+        results: MutableList<CompositeDictionary.LookupResult>,
+        successful: Int,
+        total: Int
+    ) {
+
+    }
+
+    override fun onDictionarySuggest(
+        results: MutableList<CompositeDictionary.SuggestResult>,
+        successful: Int,
+        total: Int
+    ) {
+        val m_suggestions = mutableListOf<String>()
+
+
+        if(results.size > 0){
+
+            results[0].suggestions.forEach { s ->
+                m_suggestions.add(s)
+            }
+
+        }
+
+
+        runOnUiThread {
+            // Create a new instance of your adapter with the list of suggestions
+            //val sugs = mutableListOf("apple", "banana", "cherry", "date", "elderberry")
+            val adapter = SuggestionsAdapter(m_suggestions)
+            rvSuggestions.adapter = adapter
```

Comment:
Don't create a new adapter on *every* update. This is a lot of processing overhead.

Instead, create an adapter in the activity's `onCreate()`, and update its list of items from here.

Call `adapter.notifyDataSetChanged()` whenever you update the list of items.